### PR TITLE
changes column type to uuid

### PR DESCRIPTION
**Pull request by nyashamaphosa**
## Status

* Current Status:Ready for review
* Closes https://github.com/NCCE/teachcomputing.org-issues/issues/2137

## What's changed?

*Switch column type to uuid*

### CODE REVIEW

File: spec/jobs/achiever/update_users_details_from_achiever_job_spec.rb [hunk 1]
```diff
@@ -1,11 +1,7 @@
 require 'rails_helper'
 
 RSpec.describe Achiever::UpdateUsersDetailsFromAchieverJob, type: :job do
-  let(:user) { create(:user, stem_achiever_contact_no: 'contact-valid') }
```
[the remark this answers is not in the document]
Comment (danhalson) [reply spec/jobs/achiever/update_users_details_from_achiever_job_spec.rb:4]: I think these tests are still valuable, as they off coverage that the validations are working correctly - if you change `contact-valid` to a valid uuid, they should pass?

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -1,24 +1,8 @@
...
-                                     last_name: 'Doe')
-  end
-  let(:uid) { '987654321abcDEF' }
-  let(:user) do
-    User.from_auth(uid,
-                   stem_credentials,
-                   stem_info)
-  end
+  
+  let(:user) { create(:user, stem_user_id: "id-stem_user", email: 'user@example.com')}
```
Comment (danhalson) [inline spec/models/user_spec.rb:5]: the idea of this test was the check OmniAuth was doing the correct thing with the data, by using the user factory instead we've lost that coverage, as with the test above, I think that using a proper uuid for the `achiever_contact_no` will fix whatever errors you were seeing

**Review by danhalson — CHANGES_REQUESTED** [review]
Now the test envvars have been fixed the main issue you were seeing should be sorted, then if you do the following you should get passes:

* Point `stub_valid_contact_details` back to `https://stemraspberrypiapi.dev3.smartmembership.net`
* Reinstate the removed test steps & test data
* Update test data in the specs to use a proper uuid for `stem_achiever_contact_no`

**Review by danhalson — APPROVED** [review]
Approving the non me parts of this

**Review by grega — APPROVED** [review]

**Pull request merged**